armi/utils/tabulate.py

```diff
@@ -0,0 +1,1583 @@
+
+    # add or remove an index column
+    showIndexIsSStr = type(showindex) in [str, bytes]
+    if showindex == "default" and index is not None:
+        rows = _prependRowIndex(rows, index)
+    elif isinstance(showindex, Sized) and not showIndexIsSStr:
+        rows = _prependRowIndex(rows, list(showindex))
+    elif isinstance(showindex, Iterable) and not showIndexIsSStr:
+        rows = _prependRowIndex(rows, showindex)
+    elif showindex == "always" or (_bool(showindex) and not showIndexIsSStr):
+        if index is None:
+            index = list(range(len(rows)))
+        rows = _prependRowIndex(rows, index)
+
+    # pad with empty headers for initial columns if necessary
+    headersPad = 0
+    if headers and len(rows) > 0:
+        headersPad = max(0, len(rows[0]) - len(headers))
+        headers = [""] * headersPad + headers
+
+    return rows, headers, headersPad
+
+
+def _wrapTextToColWidths(listOfLists, colwidths, numparses=True):
+    if len(listOfLists):
+        numCols = len(listOfLists[0])
+    else:
+        numCols = 0
+    numparses = _expandIterable(numparses, numCols, True)
+
+    result = []
+
+    for row in listOfLists:
+        newRow = []
+        for cell, width, numparse in zip(row, colwidths, numparses):
+            if _isnumber(cell) and numparse:
+                newRow.append(cell)
+                continue
+
+            if width is not None:
+                wrapper = TextWrapper(width=width)
+                # Cast based on our internal type handling
+                # Any future custom formatting of types (such as datetimes)
+                # may need to be more explicit than just `str` of the object
+                castedCell = (
+                    str(cell) if _isnumber(cell) else _type(cell, numparse)(cell)
+                )
+                wrapped = [
+                    "\n".join(wrapper.wrap(line))
+                    for line in castedCell.splitlines()
+                    if line.strip() != ""
+                ]
+                newRow.append("\n".join(wrapped))
+            else:
+                newRow.append(cell)
+        result.append(newRow)
+
+    return result
+
+
+def _toStr(s, encoding="utf8", errors="ignore"):
+    """
+    A type safe wrapper for converting a bytestring to str.
+
+    This is essentially just a wrapper around .decode() intended for use with things like map(), but
+    with some specific behavior:
+
+    1. if the given parameter is not a bytestring, it is returned unmodified
+    2. decode() is called for the given parameter and assumes utf8 encoding, but the default error
+       behavior is changed from 'strict' to 'ignore'
+
+    >>> repr(_toStr(b'foo'))
+    "'foo'"
+
+    >>> repr(_toStr('foo'))
+    "'foo'"
+
+    >>> repr(_toStr(42))
+    "'42'"
+
+    """
+    if isinstance(s, bytes):
+        return s.decode(encoding=encoding, errors=errors)
+    return str(s)
+
+
+def tabulate(
+    tabular_data,
```

Comment:
🐍 case -> 🐫 case

Reply:
The rest I agree with you on. 

Unnnnnfortunately...

This one this one and `tabulate_formats` below are already exposed as `snake_case` in the API.  And I checked, we have dozens of ARMI Plugins downstream that explicitly call these two out, and would need to be changed. 

THIS version of the PR means nothing downstream breaks with this PR, it's just a import change.

#shrug

Reply:
Okay, based on your opinions and @jakehader , I will just make these API changes, and make everything `camelCase`. NBD.